### ISSUE TITLE
ZEN 12.1 Maint - Unable to build some local containers.

### ISSUE THREAD

**Issue by PureOxygen-Dev** [inverse-inc/packetfence#7444]
**Describe the bug**

As discussed with @nqb I have been following the instructions in the developer guide to build patches into the running container to test new code:-

https://www.packetfence.org/doc/PacketFence_Developers_Guide.html#_applying_a_custom_patch_on_a_server_running_the_packages

Not being sure which code, is in which container I took the approach of making the changes in the code a running the build process on all the containers.

During the building of the following two containers, it results in errors and does not complete. 

**fingerbank-db**
**httpd.dispatcher**

**To Reproduce**
1. Go to host and and logon as root
2. Run command pfcmd services pf stop 
3. Run command  /usr/local/pf/addons/dev-helpers/build-local-container.sh ** ContainerName **
4. See errors below

*fingerbank-db container output*

```
/usr/local/pf/addons/dev-helpers/build-local-container.sh fingerbank-db
=================================================================================
Building image packetfence/fingerbank-db:maintenance-12-1
---------------------------------------------------------------------------------
ERRO[0005] Can't add file /usr/local/pf/logs/httpd.apache to tar: archive/tar: write too long 
ERRO[0009] Can't add file /usr/local/pf/var/proxysql/proxysql.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/api-frontend-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/collectd-unixsock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/haproxy-admin-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/haproxy-admin.stats to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/haproxy-db.stats to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/haproxy-portal-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.aaa-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.admin_dispatcher-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.dispatcher-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.portal-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.webservices-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfconfig-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfcron-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pffilter.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfperl-api-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfpki-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfsso-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/proxysql-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd-acct-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd-acct.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd-load-balancer-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd-load_balancer.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/redis_cache.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/redis_queue.sock to tar: archive/tar: sockets not supported 
Sending build context to Docker daemon  1.131GB
Step 1/5 : FROM alpine
 ---> 49176f190c7e
Step 2/5 : ARG FINGERBANK_BUILD_API_KEY
 ---> Using cache
 ---> c525b00c2856
Step 3/5 : RUN apk update && apk add curl
 ---> Using cache
 ---> 8978dac36ba8
Step 4/5 : RUN mkdir -p /usr/local/fingerbank/db
 ---> Using cache
 ---> 977ab565bba8
Step 5/5 : RUN curl --fail --retry 3 -H "Authorization: Bearer ${FINGERBANK_BUILD_API_KEY}" https://api.fingerbank.org/api/v2/download/db > /usr/local/fingerbank/db/fingerbank_Upstream.db
 ---> Running in 198797b3a7b3
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0    11    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 401
The command '/bin/sh -c curl --fail --retry 3 -H "Authorization: Bearer ${FINGERBANK_BUILD_API_KEY}" https://api.fingerbank.org/api/v2/download/db > /usr/local/fingerbank/db/fingerbank_Upstream.db' returned a non-zero code: 22
```
*httpd.dispatcher container output*

```
/usr/local/pf/addons/dev-helpers/build-local-container.sh httpd.dispatcher
=================================================================================
Building image packetfence/httpd.dispatcher:maintenance-12-1
---------------------------------------------------------------------------------
ERRO[0009] Can't add file /usr/local/pf/var/proxysql/proxysql.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/api-frontend-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/collectd-unixsock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/haproxy-admin-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/haproxy-admin.stats to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/haproxy-db.stats to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/haproxy-portal-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.aaa-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.admin_dispatcher-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.dispatcher-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.portal-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/httpd.webservices-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfconfig-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfcron-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pffilter.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfperl-api-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfpki-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/pfsso-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/proxysql-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd-acct-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd-acct.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd-load-balancer-systemd-notify.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd-load_balancer.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/radiusd.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/redis_cache.sock to tar: archive/tar: sockets not supported 
ERRO[0009] Can't add file /usr/local/pf/var/run/redis_queue.sock to tar: archive/tar: sockets not supported 
Sending build context to Docker daemon  1.131GB
Step 1/22 : ARG KNK_REGISTRY_URL
Step 2/22 : ARG IMAGE_TAG
Step 3/22 : FROM ${KNK_REGISTRY_URL}/pfbuild-debian-bullseye:${IMAGE_TAG}
 ---> 44a5b45ad493
Step 4/22 : RUN mkdir -p /usr/local/pf/
 ---> Using cache
 ---> 74000e2c4da8
Step 5/22 : WORKDIR /usr/local/pf/
 ---> Using cache
 ---> 07b7a3bef11f
Step 6/22 : COPY go/go.mod /usr/local/pf/go/
 ---> Using cache
 ---> 04322017ec46
Step 7/22 : COPY go/go.sum /usr/local/pf/go/
 ---> Using cache
 ---> 6ed2412e7147
Step 8/22 : RUN cd /usr/local/pf/go/ && go mod download
 ---> Using cache
 ---> c117a32d53ab
Step 9/22 : COPY ./go /usr/local/pf/go
 ---> Using cache
 ---> 386ed74eab51
Step 10/22 : COPY ./lib /usr/local/pf/lib
 ---> Using cache
 ---> b24b354a84d4
Step 11/22 : COPY ./html /usr/local/pf/html
 ---> Using cache
 ---> e2080a359897
Step 12/22 : COPY ./src /usr/local/pf/src
 ---> Using cache
 ---> 9755d7c30e47
Step 13/22 : COPY ./config.mk /usr/local/pf/config.mk
 ---> Using cache
 ---> 362cbd4e083e
Step 14/22 : COPY ./Makefile /usr/local/pf/Makefile
 ---> Using cache
 ---> 12511efd755e
Step 15/22 : RUN cd /usr/local/pf/go/ &&     make pfhttpd &&     make httpdispatcher
 ---> Using cache
 ---> 860f6b409f5a
Step 16/22 : RUN cd /usr/local/pf/html/common &&     make vendor  &&     make light-dist
 ---> Running in fc0db1c33b32
/bin/sh: 1: cd: can't cd to /usr/local/pf/html/common
The command '/bin/sh -c cd /usr/local/pf/html/common &&     make vendor  &&     make light-dist' returned a non-zero code: 2
```
**Expected behavior**
Both containers build without issue.

**Additional context**
For the httpd.dispatcher I can confirm the directory  /usr/local/pf/html/common does not exist.

Before attemping any code changes I made sure I was upto the latest maint 12.1 patch release.

**Comment by julsemaan** [comment]
For fingerbank-db, we're not expecting PF machines to build it although it is in our packaging.

For httpd.dispatcher, I'm guessing we need to include back html/common in our packaging since it was removed during the 12.0 release while we containerized.

@nqb, could you look at adding back html/common

**Comment by PureOxygen-Dev** [comment]
That is what I expected too. Is there a list of what service is in which container?

Also are there any items that need compiling outside of these containers when code is updated?

**Comment by nqb** [comment]
A partial fix was done (RPM + Deb): https://github.com/inverse-inc/packetfence/commit/c1d0872fec8be658931302d5ec55a91dfd716315
* RPM was not completely fixed
* Debian fix has been reverted by https://github.com/inverse-inc/packetfence/commit/a6389e81a5e974df69f6c3c5a77dbf2ccc5abbe8#diff-889c5495b68d2a0c0d1f238818716a71939918499efbc0fc6d2994319624fc67

I will adjust this.

**Comment by nqb** [comment]
Ignore my previous comment.

`html/common` is currently provided in RPM and Deb packages built against `devel` but not against `maintenance/12.1`.
I will backport fixes to 12.1 and 12.0.

**Comment by nqb** [comment]
Fix is available through latest packages on 12.1 and will be available shortly in 12.0.
